ts/Core/Utilities.ts

```diff
@@ -241,8 +220,8 @@ function merge<T>(): T {
 
     // If first argument is true, copy into the existing object. Used in
     // setOptions.
-    if (args[0] === true) {
-        ret = args[1];
+    if (typeof extendOrSource === 'boolean' && extendOrSource === true) {
```

Comment:
Isn't the `typeof` check redundant? If `extendOrSource === true`, then it follows that the type must be boolean.

```suggestion
    if (extendOrSource === true) {
```